src/app/rpmostree-builtin-status.c

```diff
@@ -498,25 +503,54 @@ status_generic (RPMOSTreeSysroot *sysroot_proxy,
               g_variant_get_child (gv_nevra_new, 1, "&s", &name_new);
               g_variant_get_child (gv_nevra_old, 1, "&s", &name_old);
 
-              if (str->len)
-                g_string_append (str, ", ");
-
               /* if pkgnames match, print a nicer version like treediff */
               if (g_str_equal (name_new, name_old))
                 {
-                  g_autofree char *old_evr = gv_nevra_to_evr (gv_nevra_old);
-                  g_autofree char *new_evr = gv_nevra_to_evr (gv_nevra_new);
-                  g_string_append_printf (str, "%s %s -> %s", name_new, old_evr, new_evr);
+                  /* let's just use str as a scratchpad to avoid excessive mallocs; the str
+                   * needs to be stretched anyway for the final output */
+                  gsize original_size = str->len;
+                  gv_nevra_to_evr (str, gv_nevra_old);
+                  g_string_append (str, " -> ");
+                  gv_nevra_to_evr (str, gv_nevra_new);
+                  const char *diff = str->str + original_size;
+                  GPtrArray *pkgs = g_hash_table_lookup (grouped_diffs, diff);
+                  if (!pkgs)
+                    {
+                      pkgs = g_ptr_array_new_with_free_func (g_free);
+                      g_hash_table_insert (grouped_diffs, g_strdup (diff), pkgs);
+                    }
+                  g_ptr_array_add (pkgs, g_strdup (name_new));
+                  g_string_truncate (str, original_size);
                 }
               else
                 {
+                  if (str->len)
+                    g_string_append (str, ", ");
+
```

Comment:
Hey, this is just a general question of mine that might not be too related to the patch. Though.. I am still not sure when/where it is the place to ask. Please correct me if I did it in the wrong place :p. 

So, what will be the case that we will have package names that are mismatched in base_local_replacements? (I guess this is the section for handling that? ).  If package names are mismatched, what output will be printed? Thank you for your time. :)

Reply:
Good question! A package can mark itself as obsoleting another package. In that case, they will not have the same name and will just have their full nevras printed out, e.g. `foo-1.0-1.x86_64 -> bar-1.3-1.x86_64`.

Reply:
Thanks for the reply, it makes sense now :)